### PR TITLE
keep track of what source line individual selectors are on

### PR DESCRIPTION
**Pull request by reedstrm**
provide enough info to create coverage tools above

### CODE REVIEW

File: cssselect2/parser.py [hunk 1]
```diff
@@ -266,14 +267,18 @@ def skip_whitespace(self):
...
-        return has_whitespace
+        return has_whitespace, self.lines
 
 
 class Selector(object):
-    def __init__(self, tree, pseudo_element=None, extensions=None):
+    def __init__(self, tree, pseudo_element=None, extensions=None, src_line=0):
         self.parsed_tree = tree
         self.extensions = extensions
+        self.source_line = src_line
```
Comment (philschatz) [inline cssselect2/parser.py]: It appears (from https://github.com/Connexions/cnx-easybake/pull/45 ) that this field is actually an offset (but maybe I am reading the code wrong); is that correct?

Comment (reedstrm) [reply cssselect2/parser.py]: yes, it's what line the selector appears in given the source as passed to cssselect2.parser.parse, based on counting `WhiteSpace` `\n`. I debated calling it something else. Granted the parser's never going to see a whole file, just a selector string. What name would you suggest? `source_line_offset`? `line_offset`?  Arguably, from the point of view of the Parser, that string _is_ the source.

Comment (philschatz) [reply cssselect2/parser.py]: 👍  for `source_line_offset` to clarify that it is an offset from the source line (or I guess the other one works too). 

💭  What if you did the offset arithmetic in the parser and just made the field be named `source_line`? Then, `source_line` always means the line in the source code and if a user wanted to calculate the offset they could do the arithmetic themselves, no?

Comment (reedstrm) [reply cssselect2/parser.py]: nope, the selector parser doesn't have the rule, just the rule.prelude, which as far is it's concerned is  an arbitrary string.

**Pull request merged**